### ISSUE TITLE
Feature Request: Output bounding boxes at 45 degree angle using Tensorflow object detection API

### ISSUE THREAD

**Issue by ajinkya933** [tensorflow/models#3397]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:windows 10
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**:1.4.0

### Describe the problem
The current version of Tensorflow object detection API is good for drawing bounding boxes at 0,90,180,360 degree respectively, namely, the output boxes we get are either vertical or horizontal. However, when there are 3 or more parallel objects: with the first object at 45 degrees to the horizontal axis and remaining two parallel to this object, the API completely ignores it. Please add a feature where the bounding box at output can tilt in accordance with the angle of image.

**Comment by bignamehyp** [comment]
Please provide more details for your problem description. Maybe a figure?

**Comment by ajinkya933** [comment]
Ok, sure. I have trained Tensorflow object-detection API to detect confinements I am using faster rcnn resnet 101 model (retraining the last layer of it) to do so. If you see the following figure: Figure 1 it detects all the confinements correctly (it also does so when there are many confinements at aligned vertically or horizontally)

Figure 1:
![capture](https://user-images.githubusercontent.com/17012391/36599713-e11d48c8-1875-11e8-8dfb-82e7c7a3de18.PNG)

However, if you see figure 2 it completely ignores confinements (my assumption is it is due to the angle of orientation of figure 2 where objects are parallel 45 degree to horizontal). 
Figure 2:
![capture1](https://user-images.githubusercontent.com/17012391/36600193-405ffd84-1877-11e8-9394-fb5779f45cda.PNG)

I want Figure 2 to output the following (note: edited in paint):
![capture2](https://user-images.githubusercontent.com/17012391/36600391-bbd6fd28-1877-11e8-8991-9e16891557c4.jpg)

Will you add a feature which does this? without rotating the image, that is outputs boxes at tilted angles in accordance with the objects?

**Comment by jkravanja** [comment]
An interesting feature would be to add "rotation" to the anchors, so it would be possible to regress not only the exact position of the bounding boxes, but also the rotation.

**Comment by srikanthinty** [comment]
Hi ajinkya, I have started working on same concept? Is there any possibility for you to share some sample code as I am completely new to Tensoflow and not able find any particular source for these kind of problems? It would be a really great help for me....

**Comment by ajinkya933** [comment]
Reopening the issue as it is not resolved.

**Comment by beric7** [comment]
This is something I am looking into as well for my research, I will post my findings after talking with my professors.

**Comment by lroberts99** [comment]
Any update on this issue?

**Comment by cannguyen275** [comment]
Still looking for this.

**Comment by Sasha1296** [comment]
Anyone find anything?

**Comment by aysark** [comment]
Subbing for updates, this would be a great feature.

**Comment by ajinkya933** [comment]
reopening the issue, its been a year the closest Ive got to solving this is using deep lab and creating a custom image segmentor

**Comment by aohan237** [comment]
@ajinkya933 
`Its been a year the closest Ive got to solving this is using deep lab and creating a custom image segmentor`
 how you do that, can you share some thoughts?
deeplab only gives you the pixel label,but you can not get  any of the angle or rotation information

**Comment by FPerezHernandez92** [comment]
Any updates on this?

**Comment by ruthvik92** [comment]
any updates on this guys? I feel like this is a very important feature request.

**Comment by ruthvik92** [comment]
> Any updates on this?

Did you figure this out?

**Comment by ameyskulkarni** [comment]
This feature will be a great improvement to the API. Do we have any updates on this or any timeline when we can expect this feature to be included?

**Comment by ajinkya933** [comment]
Reoopening it as I clearly see this as VVIMP feature request for production

**Comment by Shashwatpandey4** [comment]
I wish to work on this, and it would be really helpful if someone could guide me or collaborate with me.
Hence i would like to get assigned to this issue

**Comment by CliveKBinu** [comment]
any updates on this feature ?

**Comment by HoseinHashemi** [comment]
Keen for this feature, would be great to add the bounding box rotation to the API. 
I guess what needs to be added is another regression node for the bounding box rotation as well as the IoU calculation should be updated since with rotation the IoU will need to be calculated in a polygon manner.

**Comment by melihyazgan** [comment]
any updates on this feature ?

**Comment by Petros626** [comment]
Hey all,

This feature is still not implemented, I requested it too, but no one will code it. 

So let's come together add the algorithm for the community. BUT there's a problem... they training from scratch with rotated bounding boxes. All the models are pretrained for us users with normal bounding boxes. I don't know if the models can transfer from learning with normal bb to learn again with rot. bb.

**Comment by beric7** [comment]
If we are set on retuning a bounding box. I think an easy win would be instance segmentation + contouring. After contouring the object you could draw a box from its corner extents.
